libr/anal/var.c

```diff
@@ -604,14 +604,52 @@ static void extract_arg(RAnal *anal, RAnalFunction *fcn, RAnalOp *op, const char
 		if (type == 's') {
 			bp_off = ptr - fcn->stack;
 		}
-		char *varname = get_varname (anal, fcn, type, pfx, bp_off);
+		char *varname = NULL, *vartype = NULL;
+		if (isarg) {
+			const char *place = r_anal_cc_arg (anal, fcn->cc, ST32_MAX);
+			bool stack_rev = place ? !strncmp ("stack_rev", place, sizeof ("stack_rev")) : false;
+			char *fname = r_type_func_guess (anal->sdb_types, fcn->name);
+			int sum_sz = 0;
+			if (fname) {
+				size_t from, to, i;
+				if (stack_rev) {
+					const int cnt = r_type_func_args_count (anal->sdb_types, fname) - 1;
```

Comment:
If count is 0 it may return -1 which imho should be checked with an early if instead of going thru with this signed value. So this should be sizet